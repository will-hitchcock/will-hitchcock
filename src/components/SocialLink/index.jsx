import React from 'react'
import { string } from 'prop-types'
import Icon from '../Icon'

const SocialLink = ({ icon, url, color }) => (
  <a href={url} target="_blank" rel="noopener noreferrer">
    <Icon name={icon} color={color} size={64} />
  </a>
)

SocialLink.propTypes = {
  icon: string.isRequired,
  url: string.isRequired,
  color: string,
}

SocialLink.defaultProps = {
  // TODO: update to primary color
  color: '#000000',
}

export default SocialLink
