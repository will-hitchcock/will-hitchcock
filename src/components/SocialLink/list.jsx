import React from 'react'
import { arrayOf, shape, string } from 'prop-types'
import Row from '../Foundations/row'
import SocialLink from './index'

const SocialLinkList = ({ socialLinks }) => (
  <Row>
    {socialLinks.map((link) => (
      <SocialLink key={link.icon} {...link} />
    ))}
  </Row>
)

SocialLinkList.propTypes = {
  socialLinks: arrayOf(shape({
    icon: string.isRequired,
    url: string.isRequired,
  })).isRequired,
}

export default SocialLinkList
