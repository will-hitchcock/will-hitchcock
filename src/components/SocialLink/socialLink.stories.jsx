import React from 'react'
import { storiesOf } from '@storybook/react'
import SocialLink from './index'

storiesOf('SocialLink', module)
  .add('linkedin', () => (
    <SocialLink
      icon="linkedin"
      color="#0077B5"
      url="https://www.linkedin.com/in/willhitchcock"
    />
  ))
  .add('twitter', () => (
    <SocialLink
      icon="twitter"
      color="#1DA1F2"
      url="https://twitter.com/DrCaptScientist"
    />
  ))
  .add('github', () => (
    <SocialLink
      icon="github"
      color="#000000"
      url="https://github.com/will-hitchcock/"
    />
  ))
