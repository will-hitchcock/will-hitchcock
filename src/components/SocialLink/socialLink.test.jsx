import React from 'react'
import { shallow } from 'enzyme'
import SocialLink from './index'

describe('SocialLink', () => {
  it('renders a linkedin <SocialLink />, color #0077B5, size 32', () => {
    expect(shallow(
      <SocialLink
        icon="linkedin"
        color="#0077B5"
        url="https://www.linkedin.com/in/willhitchcock"
      />,
    )).toMatchSnapshot()
  })

  it('renders a twitter <SocialLink />, color #1DA1F2, size 64', () => {
    expect(shallow(
      <SocialLink
        icon="twitter"
        color="#1DA1F2"
        url="https://www.linkedin.com/in/willhitchcock"
      />,
    )).toMatchSnapshot()
  })

  it('renders a github <SocialLink />, color #000000, size 128', () => {
    expect(shallow(
      <SocialLink
        icon="twitter"
        color="#000000"
        url="https://www.linkedin.com/in/willhitchcock"
      />,
    )).toMatchSnapshot()
  })
})
