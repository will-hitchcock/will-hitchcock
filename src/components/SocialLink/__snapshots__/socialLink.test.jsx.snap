// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialLink renders a github <SocialLink />, color #000000, size 128 1`] = `
<a
  href="https://www.linkedin.com/in/willhitchcock"
  rel="noopener noreferrer"
  target="_blank"
>
  <Icon
    color="#000000"
    name="twitter"
    size={64}
  />
</a>
`;

exports[`SocialLink renders a linkedin <SocialLink />, color #0077B5, size 32 1`] = `
<a
  href="https://www.linkedin.com/in/willhitchcock"
  rel="noopener noreferrer"
  target="_blank"
>
  <Icon
    color="#0077B5"
    name="linkedin"
    size={64}
  />
</a>
`;

exports[`SocialLink renders a twitter <SocialLink />, color #1DA1F2, size 64 1`] = `
<a
  href="https://www.linkedin.com/in/willhitchcock"
  rel="noopener noreferrer"
  target="_blank"
>
  <Icon
    color="#1DA1F2"
    name="twitter"
    size={64}
  />
</a>
`;
